### ISSUE TITLE
Empty records or dict with indention

### ISSUE THREAD

**Issue by RDBModel** [MaybeJustJames/yaml#28]
It seems like the library only put empty line for such cases:
![image](https://user-images.githubusercontent.com/100466382/200175939-c706e5e9-1015-4549-8c18-df9ea4f41660.png)
It seems that it is better to use inline type of dict or record in such cases, like:
![image](https://user-images.githubusercontent.com/100466382/200176011-71f26260-777c-495f-868f-a06ae3015652.png)

What do you think?

**Comment by MaybeJustJames** [comment]
Hi @RDBModel! Thanks for the report. Can you tell me what you're doing exactly? Preferably with minimal example code?

**Comment by RDBModel** [comment]
Thank you for your quick response.
I have a quite big model in Elm code and converting it to the string.
There is an encoder for 'system' (that entities are on the pictures):
```
systemEncoder : System -> Encoder
systemEncoder sys =
    record
        [ ("name", string sys.name)
        , ("description", string sys.description)
        , ("relations", list relationEncoder sys.relations)
        , ("containers", dict identity containerEncoder sys.delivery)
        ]
```
later I am using `toString 2 enc` to create a string.
So encoder creates just empty lines if there is no itesm in `sys.relations` or `sys.delivery`

**Comment by MaybeJustJames** [comment]
OK, so this is the behaviour you're observing?
https://ellie-app.com/k5JJSqgv2Jda1

And you have something consuming this YAML where you need empty lists to be represented as empty lists?

My feeling is that an empty inline list is indeed more correct. But I need to check because this will be a breaking change.

**Comment by RDBModel** [comment]
Right, actually I am sending the same string (created by `toString` function) to Monaco editor and later trying to send it back to Eml. I am using encoder, like
```
systemDecoder : Decoder System
systemDecoder =
  map4 System
    (field "name" string)
    (field "description" string)
    (field "relations" (list relationDecoder))
    (field "containers" (dict deliveryDecoder))
```
but it is not parsing `system-1` properly if there are empty lines instead of `[]` or `{}` (it is basically skipping such record).

**Comment by MaybeJustJames** [comment]
Ah OK! This sounds like it isn't "round tripping". Unfortunately I don't have time to look at this right now but I absolutely will in the next day or 2.

**Comment by RDBModel** [comment]
Another case that, perhaps, can be improved https://ellie-app.com/k5NYB5sL7zDa1
![image](https://user-images.githubusercontent.com/100466382/200191616-d527a818-0c56-4eaa-a259-4939fb6da702.png)
If there is a list of records, each record will be rendered at a new line:
![image](https://user-images.githubusercontent.com/100466382/200191652-eadbb736-17a2-4a1a-b13d-2e8876569532.png)
It seems like the following looks better:
![image](https://user-images.githubusercontent.com/100466382/200191674-92a4f825-1a1c-485b-baa3-3022cf36cebd.png)

**Comment by MaybeJustJames** [comment]
> Another case that, perhaps, can be improved https://ellie-app.com/k5NYB5sL7zDa1 ![image](https://user-images.githubusercontent.com/100466382/200191616-d527a818-0c56-4eaa-a259-4939fb6da702.png) If there is a list of records, each record will be rendered at a new line: ![image](https://user-images.githubusercontent.com/100466382/200191652-eadbb736-17a2-4a1a-b13d-2e8876569532.png) It seems like the following looks better: ![image](https://user-images.githubusercontent.com/100466382/200191674-92a4f825-1a1c-485b-baa3-3022cf36cebd.png)

I agree. Could you make this a separate bug please?

**Comment by MaybeJustJames** [comment]
Hi @RDBModel ,
I've pushed a fix. I would appreciate if you could test it to check it matches your expectations please :)

**Comment by RDBModel** [comment]
@MaybeJustJames thank you very much for your work. It looks great from my perspective. Unfortunately I have no idea how to test it in my project as I don't know how to use unpublished version of the libraries in Elm (is it even possible?). As I know, Elm core team forbids it in 0.19.x version of Elm.

**Comment by MaybeJustJames** [comment]
Sorry. I was unclear. The way you would test it would be to checkout _this_ project and either use `elm repl` om some simple examples, or add a file to the `tests/` directory with anything you like that is more complete. You don't need to contribute it of course. Just convince yourself its working as you expect now.

**Comment by RDBModel** [comment]
yep, it is working as expected (at least, from my perspective). Thank you!

**Comment by ypyl** [comment]
Just checked this one again and found out that the implementation is missing a space between `[]` or `{}` and a keyword, e.g.
```
name:[]
```
but it should be 
```
name: []
```
space after the colon

**Comment by MaybeJustJames** [comment]
@ypyl and @RDBModel thanks. I've published version 2.1.3 with the encoding changes

**Comment by RDBModel** [comment]
it looks like the fix is broking encoding in the following case:
input yaml:
```yaml
domain:
  name: My domain
  actors:
    actor-1:
      name: Actor
      relations:
        - uses - system-1
  systems:
    system-1:
      name: System 1
```
The result is after decoding and decoding its back:
```yaml
domain:   name: My domain
  actors:     actor-1:
      name: Actor
      relations:         - uses - system-1
  systems:     system-1:
      name: System 1
views: {}
```
